recipes/llvm/all/conanfile.py

```diff
@@ -0,0 +1,128 @@
+from conans import ConanFile, tools, CMake
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration
+import os, shutil, glob
+
+projects = [
+    'clang',
+    'clang-tools-extra',
+    'compiler-rt',
+    'debuginfo-tests',
+    'libc',
+    'libclc',
+    'libcxx',
+    'libcxxabi',
+    'libunwind',
+    'lld',
+    'lldb',
+    'mlir',
+    'openmp',
+    'parallel-libs',
+    'polly',
+    'pstl'
+]
+
+default_projects = [
+    'clang',
+    'compiler-rt'
+]
+
+class Llvm(ConanFile):
+    name = 'llvm'
+    description = 'The LLVM Project is a collection of modular and reusable compiler and toolchain technologies'
+    url = 'https://github.com/conan-io/conan-center-index'
+    homepage = 'https://github.com/llvm/llvm-project'
+    license = 'Apache-2.0'
+    topics = 'cpp', 'compiler', 'tooling', 'clang'
+
+    settings = 'os', 'arch', 'compiler', 'build_type'
+
+    no_copy_source = True
+    _source_subfolder = 'source_subfolder'
+
+    options = {**{ 'with_' + project : [True, False] for project in projects }, **{
+        'fPIC': [True, False]
+    }}
+    default_options = {**{ 'with_' + project : project in default_projects for project in projects }, **{
+        'fPIC': True
+    }}
+    generators = 'cmake_find_package'
+
+    @property
+    def repo_folder(self):
+        return os.path.join(self.source_folder, self._source_subfolder)
+
+    def project_folder(self, project):
+        return os.path.join(self.repo_folder, project)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = 'llvm-project-llvmorg-' + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.21.3")
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, '14')
+
+    def _cmake_configure(self):
+        enabled_projects = [project for project in projects if getattr(self.options, 'with_' + project)]
+        self.output.info('Enabled LLVM subprojects: {}'.format(', '.join(enabled_projects)))
+
+        cmake = CMake(self, parallel=False);
+        cmake.configure(
+            defs = {
+                'LLVM_ENABLE_PROJECTS': ';'.join(enabled_projects),
```

Comment:
[2/2] Here:
You're adding everything as projects, and things in "compiler-rt;libc;libcxx;libcxxabi;libunwind;openmp", should be added with LLVM_ENABLE_RUNTIMES. 

Anything that's added as a RUNTIME should *not* be added as a PROJECT. 

Also, compiler-rt;libc;libcxxabi;libunwind; won't build with MSVC
cross-project-tests neither, and shouldn't be added.